corehq/apps/locations/models.py

```diff
@@ -389,10 +387,24 @@ def save(self, *args, **kwargs):
         if not self.location_id:
             self.location_id = uuid.uuid4().hex
         set_site_code_if_needed(self)
-        self.supply_point_id = sync_supply_point(self)
+        sync_supply_point(self)
         super(SQLLocation, self).save(*args, **kwargs)
         publish_location_saved(self.domain, self.location_id)
 
+    def delete(self, *args, **kwargs):
```

Comment:
i realize it may already behave this way, but is doing a hard delete of a location a desired feature? seems like some cases  or other export data still might point to that location and its information would be lost